csrc/fusion_segmenter.cpp

```diff
@@ -285,9 +286,13 @@ std::unique_ptr<SegmentedFusion> SegmentedFusion::fromCompleteFusion(
 
   // convert Welford to two-pass if option is enabled and the original heuristic
   // is persistent
+  auto isPersistentHeuristic = [&heuristic]() {
+    return heuristic == ScheduleHeuristic::InnerPersistent ||
+        heuristic == ScheduleHeuristic::Persistent ||
+        heuristic == ScheduleHeuristic::InnerOuterPersistent;
+  };
   SegmentCandidateFinderOptions scfo;
-  if (scfo.run_translate_welford &&
-      heuristic == ScheduleHeuristic::Persistent) {
+  if (scfo.run_translate_welford && isPersistentHeuristic()) {
     SegmentCandidateFinder::translateWelfordInFusion(fusion, runtime_inputs);
```

Comment:
Needs to change, otherwise the welford translate is skiped since the heuristic is `ScheduleHeuristic::InnerPersistent`